rasa_core/utils.py

```diff
@@ -601,27 +602,37 @@ def read_endpoints(cls, endpoint_file):
                 endpoint_file, endpoint_type="action_endpoint")
         model = read_endpoint_config(
                 endpoint_file, endpoint_type="models")
+        tracker_store = read_endpoint_config(
+                endpoint_file, endpoint_type="tracker_store")
 
-        return cls(nlg, nlu, action, model)
+        return cls(nlg, nlu, action, model, tracker_store)
 
-    def __init__(self, nlg=None, nlu=None, action=None, model=None):
+    def __init__(self,
+                 nlg=None,
+                 nlu=None,
+                 action=None,
+                 model=None,
+                 tracker_store=None):
         self.model = model
         self.action = action
         self.nlu = nlu
         self.nlg = nlg
+        self.tracker_store = tracker_store
 
 
 class EndpointConfig(object):
     """Configuration for an external HTTP endpoint."""
 
     def __init__(self, url, params=None, headers=None, basic_auth=None,
-                 token=None, token_name="token"):
+                 token=None, token_name="token", **kwargs):
         self.url = url
         self.params = params if params else {}
         self.headers = headers if headers else {}
         self.basic_auth = basic_auth
         self.token = token
         self.token_name = token_name
+        [setattr(
+            self, k, v) for k, v in kwargs.items() if not hasattr(self, k)]
```

Comment:
If we replace these lines with 
```
self.store_type = kwargs.pop('store_type', None)
self.kwargs = kwargs
```

then we can simplify this whole default parameter thing -> see `find_tracker_store` for the follow up

Reply:
okay, i see the other review. It will automatically be passed to the Redis or Mongo Tracker

Reply:
If we pass the extra tracker store parameters as dictionary and then pass the dictionary as key word arguments (`**kwargs`) to the actual constructor of a tracker store, then the actual tracker store should provide the default values. Was that understandable?

Reply:
Ah, you were faster 👍